src/test/java/org/jenkinsci/plugins/docker/workflow/RegistryEndpointStepTest.java

```diff
@@ -83,4 +107,117 @@
         }
     }
 
+    @Test
+    public void stepExecutionWithCredentials() throws Exception {
+        assumeNotWindows();
+        
+        IdCredentials registryCredentials = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, "registryCreds", null, "me", "pass");
+        CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), registryCredentials);
+
+        WorkflowJob p = r.createProject(WorkflowJob.class, "prj");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                        "  mockDockerLoginWithEcho {\n" +
+                        "    withDockerRegistry(url: 'https://my-reg:1234', credentialsId: 'registryCreds') {\n" +
+                        "       echo 'config would be set up to connect to https://my-reg:1234'\n" +
+                        "    }\n" +
+                        "  }\n" +
+                        "}", true));
+        WorkflowRun b = r.buildAndAssertSuccess(p);
+        r.assertLogContains("docker login -u me -p pass https://my-reg:1234", r.assertBuildStatusSuccess(r.waitForCompletion(b)));
+        r.assertLogContains("config would be set up to connect to https://my-reg:1234", b);
+    }
+
+    @Test
+    public void stepExecutionWithCredentialsAndQueueItemAuthenticator() throws Exception {
+        assumeNotWindows();
+
+        r.getInstance().setSecurityRealm(r.createDummySecurityRealm());
+        MockAuthorizationStrategy auth = new MockAuthorizationStrategy()
+                .grant(Jenkins.READ).everywhere().to("alice")
+                .grant(Computer.BUILD).everywhere().to("alice")
+                .grant(Item.CONFIGURE).everywhere().to("alice");
```

Comment:
I guess for purposes of this test you could just use `FullControlOnceLoggedInAuthorizationStrategy` or `.grant(Jenkins.ADMINISTER).everywhere().toAuthenticated()` since the ACL checks are not part of the test, unless I am misunderstanding the nature of the problem.

Reply:
The ACL is what JENKINS-48437 is about. When the build runs as a specific user, the credentials cannot be found.

```
        try (ACLContext as = ACL.as(User.getById("alice", false))) {
            b = r.buildAndAssertSuccess(p);
        }
```

This tests fails without our fix.